addons/mail/static/src/core/common/persona_model.js

```diff
@@ -144,6 +146,17 @@ export class Persona extends Record {
         return this.store.DEFAULT_AVATAR;
     }
 
+    get presenceChannel() {
+        const parts = [
+            "odoo_presence",
+            `${this.type === "partner" ? "res.partner" : "mail.guest"}_${this.id}`,
+        ];
+        if (!this.store.self.isInternalUser) {
+            parts.push(this.im_status_access_token);
+        }
```

Comment:
Could would be easier if we just always sent/validated it.
Also it's technically possible that internal users don't have access to some partners.